### PR TITLE
[TACHYON-1613] Use thread safety annotations in the minicluster module

### PR DESCRIPTION
**Pull request by jsimsa**
Addresses: https://tachyon.atlassian.net/browse/TACHYON-1613

### CODE REVIEW

File: minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java [hunk 1]
```diff
@@ -45,6 +47,7 @@
 /**
  * Local Tachyon cluster.
  */
+@NotThreadSafe
```
Comment (calvinjia) [inline minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java]: What's the convention for annotating interfaces/abstract classes & their implementations?

**Comment by calvinjia** [comment]
One high level comment, otherwise LGTM

File: minicluster/src/main/java/tachyon/underfs/UnderFileSystemCluster.java [hunk 2]
```diff
@@ -126,9 +124,14 @@ public static UnderFileSystemCluster getUnderFilesystemCluster(String baseDir,
    */
   public static boolean readEOFReturnsNegative() {
     // TODO(hy): Should be dynamically determined - may need additional method on UnderFileSystem.
-    return sUfsClz != null && sUfsClz.equals("tachyon.underfs.hdfs.LocalMiniDFSCluster");
+    return sUnderFSClass != null
+        && sUnderFSClass.equals("tachyon.underfs.hdfs.LocalMiniDFSCluster");
   }
 
+  protected String mBaseDir;
```
Comment (apc999) [inline minicluster/src/main/java/tachyon/underfs/UnderFileSystemCluster.java]: final?

**Comment by apc999** [comment]
@jsimsa  one comment left. otherwise LGTM

File: minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java (hunk 1 above)
Comment (jsimsa) [inline minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java]: No convention for interfaces yet. 

For abstract classes, the convention is that it is the contract the implementations should meet (e.g. NotThreadSafe means implementation can be NotThreadSafe or ThreadSafe, while ThreadSafe means implementations should be ThreadSafe).

File: minicluster/src/main/java/tachyon/underfs/UnderFileSystemCluster.java (hunk 2 above)
Comment (jsimsa) [inline minicluster/src/main/java/tachyon/underfs/UnderFileSystemCluster.java]: done

**Comment by jsimsa** [comment]
@apc999 @calvinjia thank you for your review ... addressed your comments, PTAL

**Comment by jsimsa** [comment]
Jenkins, test this please

File: minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java (hunk 1 above)
Comment (yupeng9) [inline minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java]: Also related, shall we have a ticket for creating a style check rule on this?

File: minicluster/src/main/java/tachyon/underfs/LocalFilesystemCluster.java [hunk 3]
```diff
@@ -22,7 +22,7 @@
 import tachyon.underfs.UnderFileSystemCluster;
 
 /**
- * The mock cluster for local file system as UnderFileSystemSingleLocal.
+ * The mock cluster for local file system.
  */
 public class LocalFilesystemCluster extends UnderFileSystemCluster {
```
Comment (yupeng9) [inline minicluster/src/main/java/tachyon/underfs/LocalFilesystemCluster.java]: annotation for this class?

Comment (jsimsa) [inline minicluster/src/main/java/tachyon/underfs/LocalFilesystemCluster.java]: done

File: minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java (hunk 1 above)
Comment (jsimsa) [inline minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java]: I would like to, but I don't see a style rule at http://checkstyle.sourceforge.net/ that we could use for that.

Comment (yupeng9) [inline minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java]: yeah, we have to implement a customized rule.

**Comment by calvinjia** [comment]
Jenkins, test this please.

**Comment by calvinjia** [comment]
LGTM

File: minicluster/src/main/java/tachyon/underfs/UnderFileSystemCluster.java (hunk 2 above)
Comment (apc999) [inline minicluster/src/main/java/tachyon/underfs/UnderFileSystemCluster.java]: I see the build error, it seems not trivial to make this `final`

**Comment by yupeng9** [comment]
LGTM

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**